nixos/lib/make-options-doc/mergeJSON.py

```diff
@@ -41,6 +41,49 @@ def unpivot(options: Dict[Key, Option]) -> Dict[str, JSON]:
         result[opt.name] = opt.value
     return result
 
+# converts in-place!
+def convertMD(options: Dict[str, Any]) -> str:
+    import mistune
+    from xml.sax.saxutils import escape
+
+    def apply(node) -> str:
+        if node['type'] == 'paragraph':
+            return "".join(map(apply, node['children']))
+        elif node['type'] == 'text':
+            return escape(node['text'])
+        elif node['type'] == 'codespan':
+            return f"<literal>{escape(node['text'])}</literal>"
+        elif node['type'] == 'block_code':
+            return f"<programlisting>{escape(node['text'])}</programlisting>"
+        elif node['type'] == 'link':
+            if node['link'][0:1] == '#':
+                attr = "linkend"
+                link = node['link'][1:]
+            else:
+                attr = "xlink:href"
+                link = escape(node['link'])
+            return f"<link {attr}=\"{link}\">{''.join(map(apply, node['children']))}</link>"
```

Comment:
[`escape`](https://docs.python.org/3/library/xml.sax.utils.html#xml.sax.saxutils.escape) is only for element bodies, not attributes. You would need to use [`quoteattr`](https://docs.python.org/3/library/xml.sax.utils.html#xml.sax.saxutils.quoteattr). Also escape `link` in both branches.